### ISSUE TITLE
Add support for self-signed certificates

### ISSUE THREAD

**Issue by leonklingele** [stevenvachon/broken-link-checker#42]
Was trying to scan a semi-private site which is using a self-signed certificate.
It would be awesome to have a flag (e.g. `--no-check-certificate`) to circumvent this:
`Error: unable to verify the first certificate`

**Comment by ajb** [comment]
A bit of googling led me to this answer: http://stackoverflow.com/questions/10888610/ignore-invalid-self-signed-ssl-certificate-in-node-js-with-https-request

tl;dr setting `NODE_TLS_REJECT_UNAUTHORIZED=0` in my environment fixed it.

**Comment by RubyGemTSL** [comment]
Thanks @ajb - I've been struggling with this!

**Comment by haveboard** [comment]
`NODE_TLS_REJECT_UNAUTHORIZED=0` isn't working for me. it attempts to start crawling the site and then doesn't report finding any links.

**Comment by arvind-d** [comment]
> `NODE_TLS_REJECT_UNAUTHORIZED=0` isn't working for me. it attempts to start crawling the site and then doesn't report finding any links.

I have the exact issue and it's so frustrating. Nothing gets crawled and it just exits without any links crawled or error messages. I believe this is due to using an SSL certificate on the site but the SSL doesn't have an alternate name for the current domain that you are trying to crawl. I believe this environment variable should have catered for this error too.

**Comment by stevenvachon** [comment]
Do you have the same issue with the v0.8.0 branch? Much has changed there.

**Comment by arvind-d** [comment]
With version 0.8.0 I get a crash:

`internal/modules/cjs/loader.js:638
    throw err;
    ^

Error: Cannot find module '../lib-cjs/cli'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
    at Function.Module._load (internal/modules/cjs/loader.js:562:25)
    at Module.require (internal/modules/cjs/loader.js:690:17)
    at require (internal/modules/cjs/helpers.js:25:18)
    at Object.<anonymous> (C:\Users\adoobary\AppData\Roaming\npm\node_modules\broken-link-checker\bin\blc:3:1)
    at Module._compile (internal/modules/cjs/loader.js:776:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:787:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)`

I tried on both Windows 10 and Linux, I get the same error.

**Comment by stevenvachon** [comment]
You first have to
```shell
npm run build
```

**Comment by stevenvachon** [comment]
Closing for now, as I think that this is fixed (on master branch).
